examples/super_resolution/main.py

```diff
@@ -45,32 +45,71 @@
 
 
 class SRDataset(torch.utils.data.Dataset):
-    def __init__(self, dataset, scale_factor):
+    def __init__(self, dataset, scale_factor, input_transform=None, target_transform=None):
         self.dataset = dataset
-        self.transform = transforms.Resize(
-            (len(dataset[0][0][0]) * scale_factor, len(dataset[0][0][0][0]) * scale_factor)
-        )
+        self.input_transform = input_transform
+        self.target_transform = target_transform
 
     def __getitem__(self, index):
-        lr_image, _ = self.dataset[index]
-        hr_image = self.transform(lr_image)
+        image, _ = self.dataset[index]
+        img = image.convert("YCbCr")
+        lr_image, _, _ = img.split()
+
+        hr_image = lr_image.copy()
+        if self.input_transform:
+            lr_image = self.input_transform(lr_image)
+        if self.target_transform:
+            hr_image = self.target_transform(hr_image)
         return lr_image, hr_image
 
     def __len__(self):
         return len(self.dataset)
 
 
-transform = transforms.Compose([transforms.ToTensor()])
+def calculate_valid_crop_size(crop_size, upscale_factor):
+    return crop_size - (crop_size % upscale_factor)
+
+
+def input_transform(crop_size, upscale_factor):
+    return Compose(
+        [
+            CenterCrop(crop_size),
+            Resize(crop_size // upscale_factor),
+            ToTensor(),
```

Comment:
@guptaaryan16 why do you crop input images instead of simply resizing them with functional API ?

Reply:
@vfdev-5 the images were of variable size between 200-300 pixels in Caltech 101 so it was giving stack error in the training without cropping them